src/adapter/breakpoints.ts

```diff
@@ -510,19 +519,7 @@ export class BreakpointManager {
 
     const wasEntryBpSet = await this._sourceMapHandlerInstalled?.entryBpSet;
     params.source.path = urlUtils.platformPathToPreferredCase(params.source.path);
-
-    // If we see we want to set breakpoints in file by source reference ID but
-    // it doesn't exist, they were probably from a previous section. The
-    // references for scripts just auto-increment per session and are entirely
-    // ephemeral. Remove the reference so that we fall back to a path if possible.
```

Comment:
This is no longer the case, and caused problems. Since a few years ago, we set the source referenced based on a hash of the URL and other information to try to make them stable between reloads, so we should leverage that information.